core/database_arango.py

```diff
@@ -932,6 +991,32 @@ def delete(self, all_versions=True):
         else:
             col = self._db.collection(self._collection_name)
         col.delete(self.id)
+        try:
+            event_type = EventType.delete
+            if self._collection_name == "tagged":
+                source_collection, source_id = self.source.split("/")
+                tag_collection, tag_id = self.target.split("/")
+                source_obj = self._db.collection(source_collection).get(source_id)
+                tag_obj = self._db.collection(tag_collection).get(tag_id)
+                event = TagEvent(
+                    type=event_type, tagged_object=source_obj, tag_object=tag_obj
+                )
+            elif self._collection_name == "links":
+                source_collection, source_id = self.source.split("/")
+                target_collection, target_id = self.target.split("/")
+                source_obj = self._db.collection(source_collection).get(source_id)
+                target_obj = self._db.collection(target_collection).get(target_id)
+                event = LinkEvent(
+                    type=event_type,
+                    source_object=source_obj,
+                    target_object=target_obj,
+                    relationship=self,
+                )
+            else:
+                event = ObjectEvent(type=event_type, yeti_object=self)
+            producer.publish_event(event)
+        except Exception:
+            logging.exception("Error while publishing event")
```

Comment:
I notice that the try blocks in this and other statements are quite large. can we narrow them down a bit? I'm assuming we're fairly confident about the code except for the publish_event part, so maybe it should just wrap that call?

Reply:
I will double check this! I just wanted to be 100% sure that code addition related to events won't break database operations